controllers/services/auth/auth_controller_actions.go

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"errors"
+	"slices"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	serviceApi "github.com/opendatahub-io/opendatahub-operator/v2/apis/services/v1alpha1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/controllers/components/dashboard"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+
+	rr.Templates = []odhtypes.TemplateInfo{
+		{
+			FS: resourcesFS,
+			Path: AdminGroupRoleTemplate,
+		},
+		{
+			FS : resourcesFS,
+			Path: AllowedGroupRoleTemplate,
+		},
+	}
+
+	return nil
+}
+
+func copyGroups(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	ai, ok := rr.Instance.(*serviceApi.Auth)
+	if !ok {
+		return errors.New("instance is not of type *services.Auth")
+	}
+
+	newGroups := false
+
+	// Get groups from the dashboardConfig
+	odhObject := &unstructured.Unstructured{}
+	odhObject.SetGroupVersionKind(gvk.OdhDashboardConfig)
+
+	if err := rr.Client.Get(ctx, client.ObjectKey{
+		Name: "odh-dashboard-config",
+		Namespace: rr.DSCI.Spec.ApplicationsNamespace,
+	}, odhObject); err == nil {
+		// We found the dashboardConfig
+		// If there is an error here we are ignoring it because we don't need the 
+		// dashboardConfig to exist
+		foundGroups, _, err := unstructured.NestedStringMap(odhObject.Object, "spec", "groupsConfig")
```

Comment:
maybe better to use the `found`  return value for this use case as the error could be returned if the input data is for any reason corrupted.

Reply:
:+1:  I've made that change.